### PR TITLE
Add typos to checks script

### PR DESCRIPTION
**Pull request by antimora**
### Checklist

- [x] Confirm that `run-checks` script has been executed.

### Changes

Add typos check installation and step in the check scripts.

### Testing

Ran the script to make sure it works.

### CODE REVIEW

File: scripts/run-checks.rs [hunk 1]
```diff
@@ -239,16 +251,33 @@ fn std_checks() {
...
+    println!("Running typos check \n\n");
+
+    // Run typos command as child process
+    let typos = Command::new("typos")
+        .stdout(Stdio::inherit()) // Send stdout directly to terminal
+        .stderr(Stdio::inherit()) // Send stderr directly to terminal
+        .spawn()
+        .expect("Failed to run typos");
+
+    // Handle rustup child process
```
Comment (Luni-4) [inline scripts/run-checks.rs:267]: Replace `rustup` with `typos`

**Review by Luni-4 — CHANGES_REQUESTED** [review]
Thanks a lot! Just some simple comments.

While at it, could you also pin `typos` ci versions? Using `sha` is not the optimal solution.

File: scripts/run-checks.rs [hunk 2]
```diff
@@ -239,16 +251,33 @@ fn std_checks() {
     burn_core_std();
 }
 
+fn check_typos() {
+    // Install typos-cli
+    cargo_install(&["typos-cli"]);
```
[the remark this answers is not in the document]
Comment (nathanielsimard) [reply scripts/run-checks.rs:256]: `cargo install` already does this, so I think it's probably faster to just call `cargo install` directly to spawn less processes.

**Comment by antimora** [comment]
> Thanks a lot! Just some simple comments.
> 
> While at it, could you also pin `typos` ci versions? Using `sha` is not the optimal solution.

I pinned to "1.16.5" version to install. If we update this value, the users will update to a new version.

File: scripts/run-checks.rs [hunk 3]
```diff
@@ -257,8 +286,9 @@ fn main() {
     {
         Some("std") => std_checks(),
         Some("no_std") => no_std_checks(),
+        Some("typos") => check_typos(),
         Some(_) | None => {
-            /* Run both "std" and "no_std" checks" */
```
[the remark this answers is not in the document]
Comment (antimora) [reply scripts/run-checks.rs:261]: Put back an updated comment.

**Review by nathanielsimard — COMMENTED** [review]
I think we have to make sure that we don't run another job just for the typos. Or update the job to use this script (less confusing).

**Review by Luni-4 — APPROVED** [review]
Fine for me!

**Comment by antimora** [comment]
@nathanielsimard OK. I have replaced the existing job with the one run from the script.

**Review by nathanielsimard — APPROVED** [review]

**Pull request merged**